tests/UnitTests/GT.Core.Tests/Services/GTCompanyServiceTests.cs

```diff
@@ -0,0 +1,561 @@
+
+			//	Act
+			var result = await sut.ExistsByNameAsync(inputCompanyName);
+
+			//	Assert
+			mockRepository.Verify(m => m.Get(), Times.Once);
+
+			result.Should()
+				.BeFalse();
+		}
+
+		[Fact]
+		public async Task GetAllAsync_ThreeEntitiesExistInDB_SucceedsAndReturnsThreeEntities()
+		{
+			//	Arrange
+			var mockLogger = new Mock<ILogger<GTCompanyService>>();
+			var mockRepository = new Mock<IGTGenericRepository<Company>>();
+
+			mockRepository
+				.Setup(m => m.Get())
+				.Returns(new List<Company>()
+				{
+					new Company(),
+					new Company(),
+					new Company(),
+				}
+				.AsQueryable().BuildMock());
+
+			var sut = new GTCompanyService(mockLogger.Object, mockRepository.Object);
+
+			//	Act
+			var result = await sut.GetAllAsync();
+
+			//	Assert
+			mockRepository.Verify(m => m.Get(), Times.Once);
+
+			result.Count.Should()
+				.Be(3);
+		}
+
+		[Fact]
+		public async Task GetAllAsync_EmptyDB_SucceedsButReturnsEmptyList()
+		{
+			//	Arrange
+			var mockLogger = new Mock<ILogger<GTCompanyService>>();
+			var mockRepository = new Mock<IGTGenericRepository<Company>>();
+
+			mockRepository
+				.Setup(m => m.Get())
+				.Returns(new List<Company>(){}
+				.AsQueryable().BuildMock());
+
+			var sut = new GTCompanyService(mockLogger.Object, mockRepository.Object);
+
+			//	Act
+			var result = await sut.GetAllAsync();
+
+			//	Assert
+			mockRepository.Verify(m => m.Get(), Times.Once);
+
+			result.Count.Should()
+				.Be(0);
+		}
+
+		[Theory]
+		[InlineData("")]
+		[InlineData(null)]
+		public async Task GetByIdAsync_InvalidCompanyId_FailsAndReturnsNull(string inputCompanyId)
+		{
+			//	Arrange
+			var mockLogger = new Mock<ILogger<GTCompanyService>>();
+			var mockRepository = new Mock<IGTGenericRepository<Company>>();
+
+			var sut = new GTCompanyService(mockLogger.Object, mockRepository.Object);
+
+			//	Act
+			var result = await sut.GetByIdAsync(inputCompanyId);
+
+			//	Assert
+			mockRepository.Verify(m => m.Get(), Times.Never);
+
+			result.Should()
+				.BeNull();
+		}
+
+		[Theory]
+		[InlineData("validCompanyId")]
+		[InlineData("    validCompanyId   ")]
+		public async Task GetByIdAsync_ValidCompanyId_SucceedsAndReturnsDTO(string inputCompanyId)
+		{
+			//	Arrange
+			var mockLogger = new Mock<ILogger<GTCompanyService>>();
+			var mockRepository = new Mock<IGTGenericRepository<Company>>();
+
+			mockRepository
+				.Setup(m => m.Get())
+				.Returns(new List<Company>()
+				{
+					new Company()
+					{
+						Id = "validCompanyId"
+					}
+				}.AsQueryable().BuildMock());
+
+			var sut = new GTCompanyService(mockLogger.Object, mockRepository.Object);
+
+			//	Act
+			var result = await sut.GetByIdAsync(inputCompanyId);
+
+			//	Assert
+			mockRepository.Verify(m => m.Get(), Times.Once);
+
+			result.Id.Should()
+				.Be("validCompanyId");
+		}
+
+		[Fact]
+		public async Task GetByIdAsync_ValidCompanyIdButNoMatchInDB_SucceedsAndReturnsNull()
+		{
+			//	Arrange
+			string inputCompanyId = "ValidInputId";
```

Comment:
Maybe we should change this to a guid as "ValidInputId" technically isn't a valid company id?